modules/mobile/app/serializers/mobile/v0/user_serializer.rb

```diff
@@ -50,54 +50,93 @@ class UserSerializer
-          residential_address: filter_keys(user.vet360_contact_info&.residential_address, ADDRESS_KEYS),
-          mailing_address: filter_keys(user.vet360_contact_info&.mailing_address, ADDRESS_KEYS),
-          home_phone_number: filter_keys(user.vet360_contact_info&.home_phone, PHONE_KEYS),
-          mobile_phone_number: filter_keys(user.vet360_contact_info&.mobile_phone, PHONE_KEYS),
-          work_phone_number: filter_keys(user.vet360_contact_info&.work_phone, PHONE_KEYS),
-          fax_number: filter_keys(user.vet360_contact_info&.fax_number, PHONE_KEYS),
+          residential_address: filter_keys(contact_info&.residential_address, ADDRESS_KEYS),
+          mailing_address: filter_keys(contact_info&.mailing_address, ADDRESS_KEYS),
+          home_phone_number: filter_keys(contact_info&.home_phone, PHONE_KEYS),
+          mobile_phone_number: filter_keys(contact_info&.mobile_phone, PHONE_KEYS),
+          work_phone_number: filter_keys(contact_info&.work_phone, PHONE_KEYS),
+          fax_number: filter_keys(contact_info&.fax_number, PHONE_KEYS),
           signin_service: user.identity.sign_in[:service_name].remove('oauth_')
         }
       end
 
-      attribute :authorized_services do |user|
-        SERVICE_DICTIONARY.filter { |_k, policies| authorized_for_service(policies, user) }.keys
+      def authorized_services
+        SERVICE_DICTIONARY.filter { |_k, policies| authorized_for_service(policies) }.keys
       end
 
-      attribute :health do |user|
+      def health
         facility_ids = user.va_treatment_facility_ids
-        facility_names = Mobile::FacilitiesHelper.get_facility_names(facility_ids)
         {
-          facilities: facility_ids.map.with_index { |id, index| facility(user, id, facility_names[index]) },
+          facilities: facility_ids.map.with_index { |id, index| facility(id, facility_names[index]) },
           is_cerner_patient: !user.cerner_id.nil?
         }
       end
 
-      def self.authorized_for_service(policies, user)
+      def authorized_for_service(policies)
         [*policies].all? { |policy| user.authorize(policy, :access?) }
       end
 
-      def self.facility(user, facility_id, facility_name)
+      def facility(facility_id, facility_name)
         cerner_facility_ids = user.cerner_facility_ids || []
         {
           facility_id: facility_id,
           is_cerner: cerner_facility_ids.include?(facility_id),
           facility_name: facility_name.nil? ? '' : facility_name
         }
       end
+
+      def fetch_additional_resources
+        Rails.logger.info("Parallel requests being made for user #{user.id} begin")
+        @facility_names, @contact_info = Parallel.map([fetch_locations, fetch_contact_info], in_threads: 2, &:call)
+        Rails.logger.info("Parallel requests being made for user #{user.id} end")
+      end
+
+      # fetches MPI, either from external source or cache, then makes an external call for facility data
+      def fetch_locations
```

Comment:
AFAICT, you can't really parallelize the MPI call and the facilities call because you have to fetch MPI to get thee facility ids.